src/nvim/tui/tui.c

```diff
@@ -374,6 +375,14 @@ static void tui_terminal_stop(UI *ui)
     ui->data = NULL;  // Flag UI as "stopped".
     return;
   }
+
+  // Wait for terminal response for background.
+  int maxwait = 20;  // 1s.
+  while (data->input.waiting_for_bg_response && maxwait > 0) {
+    maxwait -= 1;
+    loop_poll_events(data->loop, 50);
+  }
```

Comment:
Could also use a single timeout maybe?  But I was not sure what happens with more data being available etc.

Reply:
Seems good as is.

Reply:
Maybe the timeout is a bit too much though.. this might affects scripts - see the added test, where it takes 1s for nvim to quit then.
All a bit constructed of course, but maybe sending the query in the first place should/could be skipped in certain cases then automatically.
Sure, `--headless` is the way to go in scripts - I am just a bit worried to cause unnecessary slowdown/waiting.  For the use case at hand it would certainly be enough to use something like 100ms at most I guess.

Reply:
So we are adding a potential slowdown to fix a cosmetic issue?

> this might affects scripts - see the added test, where it takes 1s for nvim to quit then.

Does it affect `-es`/`-Es`? If so, we can't merge this.

> it would certainly be enough to use something like 100ms at most I guess.

Depends on system load, the buffer is received from the OS by libuv asynchronously.